panel/public/js/plugins.js

```diff
@@ -2,6 +2,7 @@ window.panel = window.panel ?? {};
 window.panel.plugins = {
 	components: {},
 	created: [],
+	headerButtons: {},
```

Comment:
It would probably make sense to rename to viewButtons here as well, right?

Reply:
I thought about it but is it too confusing that the panel extension is `viewButtons` but the config option is `headerButtons`? Cause I don't think view buttons makes sense for the config option as this is not for all views but just those 4 model views.

Reply:
I haven't thought about it before, but do we really want to limit them to model views only?

Reply:
You are right. Can we find anything else then that we could use in all places. It feels a bit weird to deal with different names.

Reply:
My mind still jumps to `model` then. But it feels a bit less connected to where they show up.

Reply:
Or we indeed open it up for all views (I mean I could see cases where it's nice to add a button on the `users` view via the config). But then our individual buttons have to get quite smart/encompass a bunch of logic. E.g. one `k-view-add-button` that includes the logic of the add buttons for users view as well as languages view. With a generic text for when used in another view context.

Reply:
The idea is growing on me. It'll take some work to transform the button components but I think in the end it could be a nice solution and the most flexible one. And then this would totally make sense term wise:

```php

return [
  'panel' => [
    'viewButtons' => [
      'page' => ['preview', 'settings', 'applause', 'status'],
      'users' => ['add', 'import']
    ]
  ]
];
```

Reply:
A `k-view-add-button` would look something like this:

```html
<template>
	<k-button
		:disabled="disabled"
		:text="text"
		icon="add"
		size="sm"
		variant="filled"
		@click="onClick"
	/>
</template>

<script>
export default {
	computed: {
		disabled() {
			if (this.$panel.view.component === "k-users-view") {
				return !this.$panel.permissions.users.create;
			}

			return false;
		},
		text() {
			if (this.$panel.view.component === "k-languages-view") {
				return this.$t("language.create");
			}

			if (this.$panel.view.component === "k-users-view") {
				return this.$t("user.create");
			}

			return this.$t("add");
		}
	},
	methods: {
		onClick() {
			if (this.$panel.view.component === "k-languages-view") {
				return this.$dialog("languages/create");
			}

			if (this.$panel.view.component === "k-users-view") {
				return this.$dialog("users/create", {
					query: {
						role: this.$panel.view.props.role?.id
					}
				});
			}
		}
	}
};
</script>
```

Reply:
@bastianallgeier this is how it could look https://github.com/getkirby/kirby/commit/f5cf0fadab5f71c7264ca0761277c68ce3d78088

I really love the flexibility of also being able to add buttons to those other views. I am not sure if I am happy enough though with the multiple add button components etc.

Reply:
I also really like it and I think it is the cleanest option in terms of naming. I would avoid confusion about the add button by simply introducing multiple buttons "add-user", "add-page", etc.

Reply:
You are right. I didn't think about the blueprints.